### DISCUSSION TITLE
Can the barrier spde model be implemented in inlabru?

### DISCUSSION THREAD

**Discussion by joenomiddlename** [inlabru-org/inlabru#104] [Q&A] [answered]
Hi inlabru community!

I am wondering if the barrier spde model of Haakon Bakka can be implemented in inlabru? I see that INLA now has a function inla.barrier.pcmatern() for generating the R-generic object. Am I able to pass this to the argument model='' when defining the components within inlabru, with main='coordinates' specified?

I remember there were some issues with implementing the barrier model in earlier inlabru versions.

Thanks!
Joe

**Comment by finnlindgren** [comment, marked as answer]
Yes. In recent versions this is now available.
In addition to supplying the rgeneric object you also need to supply the mapping information for how to link spatial coordinates to the latent variables in the barrier model, since that information is not (yet) stored in a standardised place in the rgeneric object (unlike for the the regular spde models). It should be as simple as a component specification of the form
`field(coordinates, model = barrier_model, mapper = bru_mapper(mesh))`
where `barrier_model` is the barrier model rgeneric model, based on the `mesh` geometry.

**Reply by joenomiddlename** [reply]
Thanks, Finn!!

**Reply by ASeatonSpatial** [reply]
I'm not sure anybody has tried this yet, I'd be interested to know if it worked out!
